### PR TITLE
Update the codebase to PHP 8.1

### PR DESCRIPTION
**Pull request by TavoNiievez**
I cherry-picked the commit from my other PR (#6746) to demonstrate that the tests run correctly.
I will delete it as soon as it is merged.

This PR should be merged with the Squash and merge strategy.

### CODE REVIEW

File: tests/unit/Codeception/ConfigurationTest.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function testFunctionForStrippingClassNames()
         $matches = [];
         $this->assertSame(1, preg_match('#\\\?(\w*?Helper)$#', '\\Codeception\\Module\\UserHelper', $matches));
         $this->assertSame('UserHelper', $matches[1]);
-        $this->assertSame(1, preg_match('#\\\?(\w*?Helper)$#', 'UserHelper', $matches));
+                                                        $this->assertSame(1, preg_match('#\\\?(\w*?Helper)$#', 'UserHelper', $matches));
```
Comment (Naktibalda) [inline tests/unit/Codeception/ConfigurationTest.php:39]: must be an accident :)

File: tests/unit/Codeception/Coverage/FilterTest.php [hunk 2]
```diff
@@ -95,12 +95,9 @@ public function testWhitelistIncludeFilterApplied()
 
     private function getFilterMethod(): string
     {
-        $filterMethod = 'isFiltered';
         if (method_exists($this->filter->getFilter(), 'isExcluded')) {
-            //php-code-coverage 9+
-            $filterMethod = 'isExcluded';
```
Comment (Naktibalda) [inline tests/unit/Codeception/Coverage/FilterTest.php:101]: `getFilterMethod` wrapper can be eliminated, 
 php-code-coverage 9.0 is used by PHPUnit ^9.3 and Codeception 5 has never supported anything older than 9.5.

**Review by Naktibalda — CHANGES_REQUESTED** [review]
What version is it going to be? 5.2.0? 6.0.0?

I think that 5.1 branch should stay compatible with PHP 8.0 in cases there is some critical bug that must be fixed.

**Comment by TavoNiievez** [comment]
@Naktibalda In my PR for the command folder I also had to move the minimum version of Symfony to 5.4.

I don't plan to add any breaking changes, I'm just going to remove inconsistencies in the code where I find them or rewrite small pieces of code for some simpler or more modern alternative from a Symfony or PHP version point of view.
Considering this, it can be called 5.2
People who need Symfony 4.4 or PHP 8.0 can stay on Codeception 5.1 and those who can upgrade will do so without having to change anything in their code.

File: tests/unit/Codeception/Coverage/FilterTest.php (hunk 2 above)
Comment (TavoNiievez) [reply tests/unit/Codeception/Coverage/FilterTest.php:101]: Done.

File: tests/unit/Codeception/ConfigurationTest.php (hunk 1 above)
Comment (TavoNiievez) [reply tests/unit/Codeception/ConfigurationTest.php]: yes it was, well seen, thank you.

**Review by Naktibalda — APPROVED** [review]

**Pull request merged**